easycv/resources/resources.py

```diff
@@ -57,29 +59,100 @@ def get_resource(resource_name, filename=None, show_progress=True):
     :type filename: :class:`str`, optional
     :param show_progress: True to display a progress bar, False otherwise, defaults to True
     :type show_progress: :class:`bool`, optional
-    :return: A path to the resource folder or to the given file
-    :rtype: :class:`pathlib.Path`
     """
     if resource_name in downloaded_resources():
+        info = load_resource_info(resource_name)
+        resource_type = info["type"]
+
         resource_folder = get_resources_folder() / resource_name
-        if filename is None:
-            return resource_folder
-        else:
-            file = resource_folder / filename
-            if file.is_file():
-                return file
+        if resource_type == "folder":
+            if filename is None:
+                return resource_folder
             else:
-                raise FileNotInResource(resource_name)
+                file = resource_folder / filename
+                if file.is_file():
+                    return file
+                else:
+                    raise FileNotInResource(resource_name)
+        elif resource_type == "keyvalue":
+            return read_key_values(resource_name)
+        elif resource_type == "package":
+            return __import__(info["package-name"])
     elif resource_name in available_resources():
-        print(resource_name + " is not installed. Downloading...")
-        easycv.resources.download.download_resource(
-            resource_name, show_progress=show_progress
-        )
+        info = load_resource_info(resource_name)
+        resource_type = info["type"]
+        if resource_type == "folder":
+            print(resource_name + " is not installed. Downloading...")
+            easycv.resources.download.download_resource(
+                resource_name, show_progress=show_progress
+            )
+        elif resource_type == "keyvalue":
+            print(
+                resource_name
+                + " doesn't have any values stored yet. Please insert the required data:"
+            )
+            store_key_values(resource_name, info)
+        elif resource_type == "package":
+            print(
+                "Package " + info["package-name"] + " is not installed. Downloading..."
+            )
+            install_package(resource_name, info)
         return get_resource(resource_name, filename=filename)
     else:
         raise InvalidResource(resource_name)
 
 
+def install_package(resource_name, info):
+    try:
+        if "version" in info:
+            info["pip-name"] += "==" + info["version"]
+        subprocess.check_call(
```

Comment:
Codacy found an issue: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/gh/easycv/easycv/pullRequest?prid=6261841)